src/axiselements.jl

```diff
@@ -534,6 +482,101 @@ function print_tex(io_main::IO, t::Graphics)
     end
 end
 
+########
+# Plot #
+########
+
+"""
+Default setting for the `incremental` flag of `Plot` and `Plot3`.
+"""
+const INCREMENTAL = false
+
+"Types accepted by `Plot` for the field `data`."
+const PlotData = Union{Coordinates, Table, TableFile, Expression, Graphics}
+
+"""
+$(TYPEDEF)
+
+Corresponds to the `\\addplot[3][+]` family of `pgfplot` commands.
+
+The 1-4+ argument outer constructors `Plot` and `Plot3` should be used in user
+code.
+"""
+struct Plot <: OptionType
+    is3d::Bool
+    incremental::Bool
+    options::Options
+    data::PlotData
+    trailing::AbstractVector{Any} # FIXME can/should we be more specific?
+end
+
+Plot(is3d::Bool, incremental::Bool, options::Options, data::PlotData,
+     trailing::Tuple) = Plot(is3d, incremental, options, data, collect(trailing))
+
+Base.push!(plot::Plot, element) = (push!(plot.trailing, element); plot)
+Base.append!(plot::Plot, element) = (append!(plot.trailing, element); plot)
+
+"""
+    Plot([incremental::Bool], [options::Options], data, trailing...)
```

Comment:
I agree, the boolean is kind of awkward, and I think that `Plot[3]Inc` is the best way to go forward, I will do that.

Regarding `\addlegendentry`: I realized that it is more complex than I thought, it can even take options etc, so adding a form for that, too.

I am against the keyword in `Plot` & friends, as that would break the very clear mapping between this package and `pgfplots` constructs.

Reply:
> I am against the keyword in Plot & friends, as that would break the very clear mapping between this package and pgfplots constructs.

Ok, I can respect that.